pmd-core/src/test/java/net/sourceforge/pmd/cpd/token/internal/BaseTokenFilterTest.java

```diff
@@ -0,0 +1,190 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cpd.token.internal;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ConcurrentModificationException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import org.junit.Test;
+
+import net.sourceforge.pmd.lang.TokenManager;
+import net.sourceforge.pmd.lang.ast.GenericToken;
+
+import com.google.common.collect.ImmutableList;
+
+public class BaseTokenFilterTest {
+
+    class StringToken implements GenericToken {
+
+        private final String text;
+
+        StringToken(final String text) {
+            this.text = text;
+        }
+
+        @Override
+        public GenericToken getNext() {
+            return null;
+        }
+
+        @Override
+        public GenericToken getPreviousComment() {
+            return null;
+        }
+
+        @Override
+        public String getImage() {
+            return text;
+        }
+
+        @Override
+        public int getBeginLine() {
+            return 0;
+        }
+
+        @Override
+        public int getEndLine() {
+            return 0;
+        }
+
+        @Override
+        public int getBeginColumn() {
+            return 0;
+        }
+
+        @Override
+        public int getEndColumn() {
+            return 0;
+        }
+    }
+
+    class StringTokenManager implements TokenManager {
+
+        Iterator<String> iterator = ImmutableList.of("a", "b", "c").iterator();
```

Comment:
To avoid the dependency to guava, we can also write

```suggestion
        Iterator<String> iterator = Collections.unmodifiableList(Arrays.asList("a", "b", "c")).iterator();
```